server-spi-private/src/main/java/org/keycloak/events/EventType.java

```diff
@@ -62,16 +62,24 @@ public enum EventType implements EnumWithStableIndex {
     UPDATE_EMAIL_ERROR(0x10000 + UPDATE_EMAIL.getStableIndex(), true),
     UPDATE_PROFILE(11, true),
     UPDATE_PROFILE_ERROR(0x10000 + UPDATE_PROFILE.getStableIndex(), true),
+
+    @Deprecated
```

Comment:
We should still fire those events to allow people to migrate to the changes you are introducing here. We can consider removing them in Keycloak 27/28.

Reply:
Hi @pedroigor , while I think, this is a good idea, I am unsure how to implement two events. 

Take for example [`src/main/java/org/keycloak/authentication/requiredactions/UpdateTotp.java`](https://github.com/keycloak/keycloak/blob/13826a0b63cff6bfd564a1ef2ed30a04304db259/services/src/main/java/org/keycloak/authentication/requiredactions/UpdateTotp.java#L75) which gets only a `RequiredActionContext` and the associated `EventBuilder`, but is **not** in charge of triggering the actual event (via `.success()`). The `EventBuilder` would overwrite the previously set event type and I don't see a possibility to provide two (or more) events this way, which are properly handled. Other code occurrences are quite similar to this.

Please let me know, if I am missing something, but imho this was probably a architectural design choice to only be able to have one event at the same time, which prevents a proper deprecation and migration phase for event types.

Reply:
You can fire a new event (or the existing one) after cloning the current one.

The main problem here is that we will likely break current deployments and the infrastructure they have around auditing. I'm also not sure if firing two events is a nice thing but at the same time I don't know how to not break things ...

Reply:
@pedroigor Alright, it's definitely not pretty, but now there are two events, a new generic one and and the old one.